mp2-v1/tests/common/cases/query/simple_select_queries.rs

```diff
@@ -0,0 +1,505 @@
+        .table
+        .execute_row_query(
+            &exec_query
+                .normalize_placeholder_names()
+                .to_pgsql_string_with_placeholder(),
+            &query_params,
+        )
+        .await?;
+    let matching_rows = res
+        .iter()
+        .map(|row| {
+            let key = RowTreeKey::from_bytea(row.try_get::<_, &[u8]>(0)?.to_vec());
+            let epoch = row.try_get::<_, Epoch>(1)?;
+            // all the other items are query results
+            let result = (2..row.len())
+                .filter_map(|i| {
+                    SqlType::Numeric.extract(&row, i).map(|res| match res {
+                        SqlReturn::Numeric(uint) => uint,
+                    })
+                })
+                .collect_vec();
+            Ok((key, epoch, result))
+        })
+        .collect::<Result<Vec<_>>>()?;
+    // compute input for each matching row
+    let row_tree_info = RowInfo {
+        satisfiying_rows: matching_rows
+            .iter()
+            .map(|(key, _, _)| key)
+            .cloned()
+            .collect(),
+        tree: &planner.table.row,
+    };
+    let index_tree_info = IndexInfo {
+        bounds: (planner.query.min_block, planner.query.max_block),
+        tree: &planner.table.index,
+    };
+    let current_epoch = index_tree_info.tree.current_epoch();
+    let mut matching_rows_input = vec![];
+    for (key, epoch, result) in matching_rows.into_iter() {
+        let row_proof = prove_single_row(
+            planner.ctx,
+            &row_tree_info,
+            &planner.columns,
+            epoch as BlockPrimaryIndex,
+            &key,
+            &planner.pis,
+            &planner.query,
+        )
+        .await?;
+        let (row_node_info, _, _) = get_node_info(&row_tree_info, &key, epoch).await;
+        let (row_tree_path, row_tree_siblings) = get_path_info(&key, &row_tree_info, epoch).await?;
+        let index_node_key = epoch as BlockPrimaryIndex;
+        let (index_node_info, _, _) =
+            get_node_info(&index_tree_info, &index_node_key, current_epoch).await;
+        let (index_tree_path, index_tree_siblings) =
+            get_path_info(&index_node_key, &index_tree_info, current_epoch).await?;
+        let path = RowPath::new(
+            row_node_info,
+            row_tree_path,
+            row_tree_siblings,
+            index_node_info,
+            index_tree_path,
+            index_tree_siblings,
+        );
+        matching_rows_input.push(MatchingRow::new(row_proof, path, result));
+    }
+    // load the preprocessing proof at the same epoch
+    let indexing_proof = {
+        let pk = ProofKey::IVC(current_epoch as BlockPrimaryIndex);
+        planner.ctx.storage.get_proof_exact(&pk)?
+    };
+    let pis_hash = QueryCircuitInput::ids_for_placeholder_hash(
+        &planner.pis.predication_operations,
+        &planner.pis.result,
+        &planner.query.placeholders,
+        &planner.pis.bounds,
+    )?;
```

Comment:
The reason why I added this method initially was to avoid having to pass all these inputs to `RevelationCircuitInput::new_revelation_no_results_tree`, to avoid having too many parameters in the interface of the method. So, though `new_revelation_unproven_offset` can already compute this internally (given that all these inputs need to be provided anyhow), I though to leave the computation of this outside so that there is no difference in this regard between the 2 types of queries. If you prefer to remove it, I can add these input values to `new_revelation_no_results_tree` and compute everything internally. Wdyt?

Reply:
Done in commit https://github.com/Lagrange-Labs/mapreduce-plonky2/pull/365/commits/5fd774f1b437cb511ee02f49149805ce2767dd21